cmd/siac/rentercmd.go

```diff
@@ -534,48 +553,79 @@ func renterfilesdeletecmd(path string) {
 // Downloads a path from the Sia network to the local specified destination.
 func renterfilesdownloadcmd(path, destination string) {
 	destination = abs(destination)
-	var err error
-	done := make(chan struct{})
-	go func() {
-		err = httpClient.RenterDownloadFullGet(path, destination, renterDownloadAsync)
-		close(done)
-	}()
-	downloadprogress(done, path)
+
+	// Queue the download. An error will be returned if the queueing failed, but
+	// the call will return before the download has completed. The call is made
+	// as an async call.
+	err := httpClient.RenterDownloadFullGet(path, destination, true)
+	if err != nil {
+		die("Download could not be started:", err)
+	}
+
+	// If the download is async, report success.
+	if renterDownloadAsync {
+		fmt.Printf("Queued Download '%s' to %s.\n", path, abs(destination))
+		return
+	}
+
+	// If the download is blocking, display progress as the file downloads.
+	err = downloadprogress(path)
 	if err != nil {
-		die("Could not download file:", err)
+		die("\nDownload could not be completed:", err)
 	}
 	fmt.Printf("\nDownloaded '%s' to %s.\n", path, abs(destination))
 }
 
-func downloadprogress(done chan struct{}, siapath string) {
-	time.Sleep(time.Second) // give download time to initialize
-	for {
-		select {
-		case <-done:
-			return
+// downloadprogress will display the progress of the provided download to the
+// user, and return an error when the download is finished.
+func downloadprogress(siapath string) error {
+	// If the download does not appear after a long time, give up.
+	var tryTime time.Duration
 
-		case <-time.Tick(time.Second):
-			// get download progress of file
-			queue, err := httpClient.RenterDownloadsGet()
-			if err != nil {
-				continue // benign
-			}
-			var d api.DownloadInfo
-			for _, d = range queue.Downloads {
-				if d.SiaPath == siapath {
-					break
-				}
+	for range time.Tick(OutputRefreshRate) {
+		// Get the list of downloads.
+		queue, err := httpClient.RenterDownloadsGet()
+		if err != nil {
+			continue // benign
+		}
+
+		// Search for the download in the list of downloads.
```

Comment:
This might also be a good programming task. A new endpoint that let's you fetch a single download.